kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SettingUpKolibri.vue

```diff
@@ -1,50 +1,133 @@
+        };
+      },
+
+      /** The data we will use to create the user and facility, if needed, during provisioning */
+      facilityUserData() {
+        // Data from user credentials form
+        let { full_name, username, password } = this.$store.state.onboardingData.user;
+        return {
+          full_name,
+          username,
+          password,
+          facility_name: this.$store.state.onboardingData.facility.name || null,
+          facility_dataset: {
+            learner_can_login_with_no_password: this.wizardService.state.context.requirePassword,
+          },
           extra_fields: {
-            on_my_own_setup: this.isOnMyOwnSetup(),
-            os_user: true,
+            on_my_own_setup: this.isOnMyOwnSetup,
+            os_user: this.canGetOsUser,
           },
+          auth_token: v4(),
         };
-        FacilityImportResource.createsuperuser(facilityUserData)
-          .then(
-            () => (window.location = window.urls['kolibri:kolibri.plugins.user_auth:user_auth']())
-          )
-          .catch(err => console.log(err));
+      },
+
+      /** Introspecting the machine via it's `state.context` properties */
+      isOnMyOwnSetup() {
+        return this.wizardService.state.context.onMyOwnOrGroup == UsePresets.ON_MY_OWN;
+      },
+      canGetOsUser() {
+        return this.wizardService.state.context.canGetOsUser;
+      },
+      isNewFacility() {
+        return this.wizardService.state.context.facilityNewOrImport === FacilityTypePresets.NEW;
+      },
+      isLearnOnlyDevice() {
+        return this.wizardService.state.context.fullOrLOD === DeviceTypePresets.LOD;
+      },
+    },
+    mounted() {
+      if (this.isOnMyOwnSetup) {
+        // 1) On my own
+        if (this.canGetOsUser) {
+          // 1.1) If we can get the OS user, we'll do this
+          this.createAndProvisionOnMyOwnUserApp();
+        } else {
+          // 1.2) If user is doing "on my own" setup but we cannot get OS user
+          this.createAndProvisionOnMyOwnUserDevice();
+        }
+      }
+
+      // From here, all are going to be "group learning" flows
+      if (!this.isLearnOnlyDevice) {
+        // 2) Group learning, Full device setup
+        if (this.isNewFacility) {
+          // 2.1) New Facility
+          this.createAndProvisionNewFullFacilityDevice();
+        } else {
+          // 2.2) Import Facility
+          // We already have the facility imported, just provision and redirect
+          this.provisionDevice();
+        }
       } else {
-        window.location = window.urls['kolibri:kolibri.plugins.user_auth:user_auth']();
+        // 3) Group learning, learn only device
+        this.provisionDevice();
       }
     },
     methods: {
-      isOnMyOwnSetup() {
-        return this.wizardService.state.context.onMyOwnOrGroup == UsePresets.ON_MY_OWN;
+      createAndProvisionOnMyOwnUserDevice() {
+        SetupWizardResource.createonmyownuser(this.facilityUserData).then(() =>
```

Comment:
I don't understand why the user creation and provisioning steps have been separated in these three instances, when they are then just chained without any intermediary state? This could simply use the existing endpoint without any issues, and one fewer request.